lib/zstream/protocol.ex

```diff
@@ -17,12 +19,28 @@ defmodule Zstream.Protocol do
         Extra.zip64_extended_info(0, 0, local_file_header_offset)
       )
 
+    encryption_coder = get_encryption_coder(options)
+
+    encryption_extra_field =
+      if encryption_coder && function_exported?(encryption_coder, :extra_field_data, 1) do
+        encryption_coder.extra_field_data(options)
+      else
+        <<>>
+      end
```

Comment:
could directly update extra_filed here, so we don't have to build list twice later